Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -309,15 +313,16 @@ private class FindAndModifyOptions {
             }
         }
 
-        if extra.keys.count > 0 && !mongoc_find_and_modify_opts_append(self._options, extra.data) {
+        let appendSuccess = extra.keys.count <= 0 || mongoc_find_and_modify_opts_append(self._options, extra.data)
```

Comment:
AFAIK, `Document.keys` returns `[String]`. In this case, would it make more sense to use `isEmpty` rather than `<= 0`? I imagine `<=` was probably used to be defensive, but I think we're only checking whether or not there are any keys at all.

This might also be clearer if written like the blocks above:

```
if !extra.keys.isEmpty {
    guard mongoc_find_and_modify_opts_append(...) else {
        throw ...
    }
}
```

Alternatively, we can craft a single guard statement:

```
guard extra.keys.isEmpty || mongoc_find_and_modify_opts_append(...) else {
   throw ...
}
```

In this case, `mongoc_find_and_modify_opts_append()` would never be called if the keys were empty due to short-circuiting.

I'm happy to defer to @kmahar as to which is more Swifty. The first is a bit more verbose but is probably the most readable.

Reply:
lol oops didn't see your comment came in as I was writing mine :) 

I'm fine with either suggestion of yours, so long as we use `isEmpty` directly on the doc rather than on the keys array, for the reasons explained in my above comment.

Reply:
I've opened [SWIFT-229](https://jira.mongodb.org/browse/SWIFT-229) about looking for these cases throughout the whole codebase

Reply:
@jmikola Yup, this is waiting on PR#126 from Kaitlin. So maybe feel free to resolve this thread and we can use Kaitlin's comment below as the thread for this line of the PR?

Reply:
#126 merged